includes/custom_event_data_not_fresh.js

```diff
@@ -0,0 +1,28 @@
+module.exports = (params) => {
+  return params.customEventSchema
+    // Only generate assertions for custom events which have dataFreshnessDays configured
+    .filter(customEvent => customEvent.dataFreshnessDays)
+    // Generate a separate assertion with a different name for each eventtype_eventsourcename custom event called something like eventtype_custom_event_data_not_fresh_eventsourcename
+    .forEach(customEvent => {
+      if (!Number.isInteger(customEvent.dataFreshnessDays) || customEvent.dataFreshnessDays < 1) {
+        throw new Error(`dataFreshnessDays parameter for the ${customEvent.eventType} eventType is not a positive integer.`);
+      } else if (customEvent.dataFreshnessDisableDuringRange && params.disableAssertionsNow) {

```

Comment:
Note to AK - Check how dataFreshnessDisableDuringRange and disableAssertionsNow work to ensure && is correct here. 

Does dataFreshnessDisableDuringRange set disableAssertionsNow when in a range?

Reply:
@Arjun-Kharbanda yes it does! See https://github.com/DFE-Digital/dfe-analytics-dataform/blob/771fda76a997cf68bd24027643c618e7722dc754/includes/parameter_functions.js#L136C4-L136C65